### PR TITLE
Improve host to device memory copy

### PR DESCRIPTION
**Pull request by okuta**
This PR removes CPU overhead.

### CODE REVIEW

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -2124,20 +2125,20 @@ cpdef ndarray array(obj, dtype=None, bint copy=True, str order='K',
             order = 'A'
         a_cpu = numpy.array(obj, dtype=dtype, copy=False, order=order,
                             ndmin=ndmin)
-        order = 'C' if a_cpu.flags.c_contiguous else 'F'
+        order = None if a_cpu.flags.c_contiguous else 'F'
```
Comment (kmaehashi) [inline cupy/core/core.pyx]: Is this change needed? It seems the behavior does not change between `'C'` and `None`.

Comment (okuta) [reply cupy/core/core.pyx]: `None` means `'C'`. And, checking `None` is faster than  checking `'C'`.

Comment (kmaehashi) [reply cupy/core/core.pyx]: I see, thanks!

**Comment by kmaehashi** [comment]
@okuta Could you check Jenkins test failures?

**Comment by okuta** [comment]
Cloud you merge #1385 first?

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
@okuta Could you check #1393 to fix jenkins failure?

**Comment by kmaehashi** [comment]
Jenkins, test this please.

File: cupy/core/core.pyx [hunk 2]
```diff
@@ -2139,24 +2140,26 @@ cpdef ndarray array(obj, dtype=None, bint copy=True, str order='K',
                 a = a.view()
             a.shape = (1,) * (ndmin - ndim) + a.shape
     else:
-        if order == 'K':
-            order = 'A'
+        if order is not None and len(order) == 1 and order in 'KAka':
```
Comment (kmaehashi) [inline cupy/core/core.pyx]: Order needs to be normalized to accept longer names (`ANY`, `KEEP`).

**Comment by kmaehashi** [comment]
I benchmarked this PR with [this code](https://github.com/cupy/cupy-benchmark/pull/10) and confirmed performance improvement! 🎉 

Current master:

```
====== ========== ======== ======== ========
--              in_order / out_order
------ -------------------------------------
 ndim    C / C     C / F    F / C    F / F
====== ========== ======== ======== ========
  1      1.53ms    1.47ms   1.46ms   1.40ms
  2      1.44ms    1.47ms   1.51ms   1.43ms
  5      1.49ms    3.78ms   2.34ms   2.91ms
  8     750.61ms   4.62s    1.36s    4.04s
====== ========== ======== ======== ========
```

This PR:

```
====== ========== ======== ======== ==========
--               in_order / out_order         
------ ---------------------------------------
 ndim    C / C     C / F    F / C     F / F   
====== ========== ======== ======== ==========
  1      1.31ms    1.30ms   1.37ms    1.29ms  
  2      1.41ms    1.47ms   1.34ms    1.40ms  
  5      1.47ms    2.38ms   2.35ms    1.55ms  
  8     594.14ms   1.08s    2.39s    635.28ms 
====== ========== ======== ======== ==========
```

**Comment by okuta** [comment]
Thank you for your performance tests.
I fixed.

**Comment by kmaehashi** [comment]
Thanks! Jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM!
